src/corehost/cli/setup.cmake

```diff
@@ -11,6 +11,12 @@ if(WIN32)
     add_compile_options($<$<CONFIG:Release>:-DNDEBUG>)
     add_compile_options($<$<CONFIG:RelWithDebInfo>:-DNDEBUG>)
     add_compile_options($<$<CONFIG:Debug>:/Od>)
+    add_compile_options(/guard:cf) 
+    add_compile_options(/d2Zi+) # make optimized builds debugging easier
+    add_compile_options(/Oi) # enable intrinsics
+    add_compile_options(/Oy-) # disable suppressing of the creation of frame pointers on the call stack for quicker function calls
+    add_compile_options(/GF) # enable read-only string pooling
+    add_compile_options(/FC) # use full pathnames in diagnostics
```

Comment:
Those are added in the component binaries in their CMakeLists.txt pls. see 

https://github.com/dotnet/cli/blob/rel/1.0.0/src/corehost/cli/CMakeLists.txt#L9

https://github.com/dotnet/cli/blob/rel/1.0.0/src/corehost/cli/fxr/CMakeLists.txt#L9

https://github.com/dotnet/cli/blob/rel/1.0.0/src/corehost/cli/dll/CMakeLists.txt#L9